docs/adrs/0002-dedicated-timdex-record-field-methods.md

```diff
@@ -0,0 +1,101 @@
+
+### 1- Refactor the base `Transformer` class to have all `TimdexRecord` fields as dedicated methods
+
+At the time of writing, the `TimdexRecord` has 30 fields:
+
+  * REQUIRED
+    * `source`
+    * `source_link`
+    * `timdex_record_id`
+    * `title`
+
+  * OPTIONAL
+    * `alternate_titles`
+    * `call_numbers`
+    * `citation`
+    * `content_type`
+    * `contents`
+    * `contributors`
+    * `dates`
+    * `edition`
+    * `file_formats`
+    * `format`
+    * `funding_information`
+    * `holdings`
+    * `identifiers`
+    * `languages`
+    * `links`
+    * `literary_form`
+    * `locations`
+    * `notes`
+    * `numbering`
+    * `physical_description`
+    * `publication_frequency`
+    * `publication_information`
+    * `related_items`
+    * `rights`
+    * `subjects`
+    * `summary`
+
+The base `Transformer` class will have a dedicated method for each of these 30 `TimdexRecord` fields, providing a clear and modular approach to defining logic for setting those fields. 
+
+The required fields will be an `@abstractmethod`, meaning any extending classes will be required to define them.
+
+The rest of the optional fields will receive an `@optional_field("<FIELD_NAME>")` decorator that will identify them as a) methods meant to parsing data for a given field, and b) the name of that field.  Using a decorator provides a few of benefits:
+
+1. It can be programmatically determined which transformations define logic, for which fields, even into inheritance
+2. Hard-coding method-to-field invocations, or using method naming conventions, are not needed
+3. Decorators could eventually pass more information to methods if needed (e.g. state)
+4. Self-documenting that methods have an explicit purpose
+
+From there, transformations will extend `Transformer` and override methods for specific fields, with logic specific to that transformation.  Multiple inheritance is also simplified in this approach, where extending classes continue to refine the behavior for a specific target field.
+
+### 2- Refactor pre-existing transformations to use this new approach
+
+With the new structure in place, refactor pre-existing source transformations to use it.  This may include revisiting some of the transformation inheritance if there is an opportunity for intermediate transformation classes now that an intermediate class could theoretically handle only one, two, or a handful of fields. 
+
+The end goal should be transformations that are identical to the original versions, only the structure of the transformations will have changed in this first pass.
+
+### 3- Refactor tests to unit test the field methods where possible
+
+With dedicated methods for each field, it will be possible to refactor the tests to test only how that method behaves.  
+
+For example, given source `A` a test could confirm that given an input XML record, it returns an expected value for the `TimdexRecord.dates` field by calling the method registered with `@optional_field("dates")`.
```

Comment:
Small note: I would add a comma after "source `A`". 🤓

Reply:
Also love this capability! Do you foresee a unit test for field methods for _every_ source that gets added to `transmogrifier`? 🤔

Reply:
I think that's a reasonable expecatation?  If there is one unit test per source, per field (naming convention practically writes itself), you could rest assured the final record would be a combination of them.  Of course, a "full transform" test is probably still a good idea.

But breaking them into individual tests, would allow you to test variations all in that same unit test.

E.g. test `test_source_libguides_field_method_dates` could test:
  * values we expect
  * bad values, throw errors
  * etc.